src/main/java/com/powsybl/openloadflow/ac/TransformerVoltageControlOuterLoop.java

```diff
@@ -35,11 +37,12 @@ public OuterLoopStatus check(OuterLoopContext context, Reporter reporter) {
 
         if (context.getIteration() == 0) {
             for (LfBus bus : context.getNetwork().getBuses()) {
-                if (bus.isDiscreteVoltageControlled()) {
+                Optional<DiscreteVoltageControl> dvc = bus.getDiscreteVoltageControl().filter(dvc0 -> bus.isDiscreteVoltageControlled());
+                if (dvc.isPresent()) {
```

Comment:
Can you hare change for a more explicit name such as `voltageControl` because you use it during several lines?

Reply:
in fact I changed the code to have something similar to `PhaseControlOuterLoop`, which is clearer at the cost of creating a list of enabled voltage controls.